File: fiyaback/src/main/java/com/app/fiya/user/dto/UserResponse.java
package com.app.fiya.user.dto;

import com.app.fiya.MyPage;
import com.app.fiya.field.dto.FieldFavResponse;
import com.app.fiya.field.dto.FieldListResponse;
import com.app.fiya.user.model.User;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.RequiredArgsConstructor;
import lombok.experimental.SuperBuilder;

import java.time.LocalDate;
import java.util.Set;
import java.util.stream.Collectors;

@Data
@AllArgsConstructor
@RequiredArgsConstructor
@SuperBuilder
public class UserResponse {

    protected String id;
    protected String image;
    protected String email;
    protected String name;
    protected LocalDate birthdate;
    protected Set<FieldFavResponse> favourites;
    protected boolean isOnTeam;
    protected String dni;
    protected String rol;

    public static UserResponse fromUser(User user) {

        return UserResponse.builder()
                .id(user.getId().toString())
                .image(user.getImage())
                .email(user.getEmail())
                .name(user.getName())
                .birthdate(user.getBirthdate())
                .dni(user.getDni())
                .rol(user.getRole().toString())
                .favourites(user.getFavourites().stream().map(FieldFavResponse::of).collect(Collectors.toSet()))
                .isOnTeam(user.isOnTeam())
                .build();
    }
}

File: fiya_front/lib/models/user/user_response.dart
class UserResponse {
  String? id;
  String? image;
  String? email;
  String? name;
  String? birthdate;
  List<int>? favourites;
  String? dni;
  String? rol;

  UserResponse(
      {this.id,
      this.image,
      this.email,
      this.name,
      this.birthdate,
      this.favourites,
      this.dni,
      this.rol});

  UserResponse.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    image = json['image'];
    email = json['email'];
    name = json['name'];
    birthdate = json['birthdate'];
    favourites = json['favourites'].cast<int>();
    dni = json['dni'];
    rol = json['rol'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['id'] = this.id;
    data['image'] = this.image;
    data['email'] = this.email;
    data['name'] = this.name;
    data['birthdate'] = this.birthdate;
    data['favourites'] = this.favourites;
    data['dni'] = this.dni;
    data['rol'] = this.rol;
    return data;
  }
}
